### PR TITLE
fix(mapping-saturation): add missing edge case detection for too few data points.

### LINKED ISSUE #507: Internal Server Error when requesting the Mapping Saturation Indicator with to few data points
Logs:

```
  File "/opt/oqt/./ohsome_quality_analyst/oqt.py", line 244, in _
    indicator.calculate()
  File "/opt/oqt/./ohsome_quality_analyst/indicators/mapping_saturation/indicator.py", line 111, in calculate
    self.fitted_models = self.select_models(fitted_models)
  File "/opt/oqt/./ohsome_quality_analyst/indicators/mapping_saturation/indicator.py", line 211, in select_models
    avg_last_2_years = np.sum(self.values[-24]) / 24
IndexError: list index out of range
```

Request: custom `layer` parameter

```json
{"name": "Cycleways", "description": "Number of features taged as `highway=cycleway`.", "data": {"result": [{"timestamp": "2014-01-01T00:00:00Z", "value": 432.0}, {"timestamp": "2015-01-01T00:00:00Z", "value": 461.0}, {"timestamp": "2016-01-01T00:00:00Z", "value": 362.0}, {"timestamp": "2017-01-01T00:00:00Z", "value": 365.0}, {"timestamp": "2018-01-01T00:00:00Z", "value": 438.0}, {"timestamp": "2019-01-01T00:00:00Z", "value": 452.0}, {"timestamp": "2020-01-01T00:00:00Z", "value": 486.0}, {"timestamp": "2021-01-01T00:00:00Z", "value": 766.0}, {"timestamp": "2022-01-01T00:00:00Z", "value": 2036.0}]}}
```

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description

- fix(mapping-saturation): add missing edge case detection for too few data points.
- test(mapping-saturation): add and rewrite tests
    - add tests for checking edge cases
    - rewrite tests to use pytest instead of unittest
    - introduce new session wide fixtures in conftest

### Corresponding issue
Closes #507  

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,11 @@
 - change indicator and report keys from lower camel case to lower hyphen ([#398])
 - rename API parameter `layerKey` and `layer` to `topic` ([#501])
 
+### Bug Fixes
+
+- fix(mapping-saturation): add missing edge case detection for too few data points. ([#512])
+- fix(mapping-saturation): allow result value above 100% ([#479])
```
Comment (joker234) [inline CHANGELOG.md:14]:
```suggestion
- mapping-saturation: add missing edge case detection for too few data points. ([#512])
- mapping-saturation: allow result value above 100% ([#479])
```
Or if you want more pretty to read names:
```suggestion
- Mapping Saturation indicator: add missing edge case detection for too few data points. ([#512])
- Mapping Saturation indicator: allow result value above 100% ([#479])
```

File: workers/tests/integrationtests/indicators/test_mapping_saturation.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    MappingSaturation,
+)
+from tests.integrationtests.utils import oqt_vcr
+
+
+class TestCheckEdgeCases:
+    @pytest.fixture()
+    def indicator(self, topic_building_count, feature_germany_heidelberg):
+        i = MappingSaturation(topic_building_count, feature_germany_heidelberg)
+        return i
```
Comment (joker234) [inline workers/tests/integrationtests/indicators/test_mapping_saturation.py:17]:
```suggestion
        return MappingSaturation(topic_building_count, feature_germany_heidelberg)
```

File: workers/tests/integrationtests/indicators/test_mapping_saturation.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    def test_not_enough_data_points(self, indicator):
+        indicator.values = [0, 1, 10]
+        assert indicator.check_edge_cases() == (
+            "Not enough data points available in this regions. "
+            + "The Mapping Saturation indicator needs data for at least 36 months."
+        )
+
+    def test_data_deleted(self, indicator):
+        indicator.values = [i for i in range(36)]
+        indicator.latest_value = 0
```
Comment (joker234) [inline workers/tests/integrationtests/indicators/test_mapping_saturation.py:39]: You could remove the `latest_value` implementation in favour of `values[-1]`

Comment (matthiasschaub) [reply workers/tests/integrationtests/indicators/test_mapping_saturation.py:38]: Done

**Review by joker234 — APPROVED** [review]

**Review by joker234 — APPROVED** [review]

**Review by joker234 — APPROVED** [review]
